File: CS/test6.js
const nearest_power_of_two = function (n) {
  let x = 0 ; 
  while (Math.pow(2,x) <= n) {
    // console.log(x, Math.pow(2,x));
    x++; 
  }
  return Math.pow(2,x-1); 
} 

// console.log( nearest_power_of_two(5));
// console.log( nearest_power_of_two(10));
// console.log( nearest_power_of_two(2)); 
// console.log( nearest_power_of_two(100)); 

File: CS/test6.rb
def nearest_power_of_two(n) 
   x = 0 
  while ((2 ** x) <= n) do 
    x = x + 1 
  end 
   2** (x-1); 
  
end 



# answer: 8, 2, 64 
puts( nearest_power_of_two(5))
puts ( nearest_power_of_two(10))
puts( nearest_power_of_two(2))
puts( nearest_power_of_two(100))